datafusion/core/Cargo.toml

```diff
@@ -104,17 +104,22 @@ xz2 = { version = "0.1", optional = true }
 [dev-dependencies]
 arrow = { version = "31.0.0", features = ["prettyprint", "dyn_cmp_dict"] }
 async-trait = "0.1.53"
+bigdecimal = "0.3.0"
 criterion = "0.4"
 csv = "1.1.6"
 ctor = "0.1.22"
 doc-comment = "0.3"
 env_logger = "0.10"
+half = "2.2.1"
 parquet-test-utils = { path = "../../parquet-test-utils" }
+postgres-types = { version = "0.2.4", features = ["derive", "with-chrono-0_4"] }
 rstest = "0.16.0"
+rust_decimal = { version = "1.27.0", features = ["tokio-pg"] }
 sqllogictest = "0.10.0"
 test-utils = { path = "../../test-utils" }
+testcontainers = "0.14.0"
 thiserror = "1.0.37"
-
+tokio-postgres = "0.7.7"
```

Comment:
Here are the reasons for these dependencies:
- `half` - `f16` type for Datafusion
- `testcontainers` - creates a fresh docker container with Postgres for each sqllogictest file.
- `postgres-types` and `tokio-postgres` - these are required for writing a Postgres client
- `rust_decimal` - converts Postgres "numeric" type to a rust type
- `bigdecimal` - provides a common type to do floating number rounding. `rust_decimal`, unfortunately, doesn't handle numbers of arbitrary precision. For example, `rust_decimal` could not parse `26156334342021890000000000000000000000` that is currently present in one of `.slt` tests in Datafusion.